components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -16405,8 +16405,15 @@ public void addDeployedAPIRevision(String apiRevisionId, List<DeployedAPIRevisio
                     connection.commit();
                 } catch (SQLException e) {
                     connection.rollback();
-                    handleException("Failed to add deployed API Revision for Revision UUID "
-                            + apiRevisionId, e);
+                    // handle concurrent db entry update. Fix duplicate primary key issue.
+                    if (e.getMessage().toLowerCase().contains("primary key violation") ||
+                            e.getMessage().toLowerCase().contains("duplicate entry")) {
+                        log.warn("Duplicate entries detected for Revision UUID " + apiRevisionId +
```

Comment:
How about making this as a DEBUG log entry? 
We know this is expected and these WARN logs will most probably be noisy.. 
@tharindu1st @sumedhassk WDYT?

Reply:
+1 Fixed with https://github.com/wso2/carbon-apimgt/pull/10720/commits/6c3b2427436375aff7328fa8a34cd9c0b9038fea